Source/WebKit/Shared/Cocoa/CoreIPCDictionary.mm

```diff
@@ -50,13 +51,76 @@
     }
 }
 
-RetainPtr<id> CoreIPCDictionary::toID() const
+bool CoreIPCDictionary::keyHasValueOfType(const String& key, IPC::NSType type) const
+{
+    createNSDictionaryIfNeeded();
+    id object = [m_nsDictionary.get() objectForKey:(NSString *)key];
+    if (!object)
+        return false;
+    return IPC::typeFromObject(object) == type;
+}
+
+bool CoreIPCDictionary::keyIsMissingOrHasValueOfType(const String& key, IPC::NSType type) const
+{
+    createNSDictionaryIfNeeded();
+    id object = [m_nsDictionary.get() objectForKey:(NSString *)key];
+    if (!object)
+        return true;
+    return IPC::typeFromObject(object) == type;
+}
+
+bool CoreIPCDictionary::collectionValuesAreOfType(const String& key, IPC::NSType targetType) const
 {
-    auto result = adoptNS([[NSMutableDictionary alloc] initWithCapacity:m_keyValuePairs.size()]);
-    for (auto& keyValuePair : m_keyValuePairs)
-        [result setObject:keyValuePair.value->toID().get() forKey:keyValuePair.key->toID().get()];
+    createNSDictionaryIfNeeded();
+    NSArray *object = [m_nsDictionary.get() objectForKey:(NSString *)key];
+    if (!object)
+        return true;
 
-    return result;
+    if (![object isKindOfClass:[NSArray class]])
+        return false;
+
+    for (id value in object) {
+        if (IPC::typeFromObject(value) != targetType)
+            return false;
+    }
+
+    return true;
+}
+
+bool CoreIPCDictionary::collectionValuesAreOfType(const String& key, IPC::NSType keyType, IPC::NSType valueType) const
+{
+    createNSDictionaryIfNeeded();
+    NSDictionary *object = [m_nsDictionary.get() objectForKey:(NSString *)key];
+    if (!object)
+        return true;
+
+    if (![object isKindOfClass:[NSDictionary class]])
```

Comment:
I'm preferring dot syntax like NSDictionary.class more and more, but it doesn't really matter.

Reply:
I'm not against changing this. I use it sometimes, too (I think this was just a copy/paste from somewhere else.)